accepted/2021/system-drawing-win-only/system-drawing-win-only.md

```diff
@@ -0,0 +1,193 @@
+>   `libgdiplus` separately. For more information, see Install .NET on Linux or
+>   Install .NET on macOS.
+>
+>   If you can't use System.Drawing with your application, recommended
+>   alternatives include ImageSharp, SkiaSharp, and Windows Imaging Components.
+
+However, this disclaimer doesn't feel clear enough to steer people towards using
+other graphics libraries.
+
+The proposal is to do the following for .NET 6:
+
+1. **Mark the assembly as only being supported on Windows**. That will cause
+   the platform compatibility analyzer to warn when consumed from a project
+   that isn't targeting net6.0-windows (or marked as Windows-specific).
+
+2. Throw `PlatformNotSupportedException` when trying to load `libgdiplus`
+   pointing to an aka.ms link with the reasons behind it and recommending
+   stable, well-maintained open-source library alternatives
+
+3. Add targets to the package that when it is restored using a non-windows RID,
+   it warns that we no longer support it on non-windows and point to the aka.ms
+   link with more info.
+
+The challenge is that `System.Drawing.Common` is used by 1st party consumers
+such as ML.NET, which uses it for image manipulation and even exposes an API
+that returns a Bitmap. At the same time, https://github.com/dotnet/iot is in the process
+of moving to use ImageSharp and so far they had a great migration experience. We
+need to engage with 1st party customers to see what impact this proposal will
+have on them.
+
+***Note:** This only affect `System.Drawing.Common` and would leave
+`System.Drawing.Primitives` as-is. That assembly contains primitive types
+that don't depend on GDI+, such `Rectangle`, `Point`, and `Size`. Any types in `System.Drawing.Common`
+that don't depend on GDI+, will be moved to `System.Drawing.Primitives`.*
+
+# Q & A
+
+## Will this prevent applications to move to .NET 6?
+
+No, `System.Drawing.Common` is a standalone package that does not ship as part
+of the shared framework. For that reason if an application wants to keep using
+`System.Drawing.Common` on `Unix` and move to .NET 6, they can reference the
+`5.0.x` version of the package.
+
+## How will applications migrate to recommended libraries?
+
+We will provide with the right documentation via a breaking change notice and a document
+that will contain guidance and samples.
+
+## What does the usage data of `System.Drawing.Common` reveal?
+
+In order to assess how `System.Drawing.Common` we looked at the API usage from
+libraries published to nuget.org.
+
+We looked at member refs to types that begin with `System.Drawing.` and
+aggregated the hits by type and member name. We also pulled in the TFM of the
+assemblies in question to help categorize the hits. There's a lot of data (there
+are a lot of members * TFMs), so instead of just huge raw tables of nonsense,
+we'll provide a summary of what we observed. 
+
+If we look at .NET Core family of TFMs, the top APIs are:
```

Comment:
Was there any feasible way to guess at Unix vs Windows usage levels - RID specific libraries that depend on them? @marklio

Reply:
Unfortunately there is not as NuGet is mostly libraries and we don't know who consumes them and how, but from the APIs we see are mostly used on NuGet we can see that this is definitely replaceable by the other libraries mentioned in this doc.

Reply:
Right, it is exceedingly rare for libraries in NuGet to target RIDs, and even more rare for that to be leveraged as a means to constrain the applicability of a library to an OS. In my experience, targeting RIDs typically means that you're providing a platform abstraction layer of sorts as an implementation detail of a cross-plat API rather than constraining a library to a particular platform. And again, it's so rare that it has no predictive power for what we're trying to determine.

Reply:
Got it.

Reply:
Right. Also, we're looking at libraries, not applications. Even if there were RIDs, this still wouldn't tell us what consumers are using. All it would tell us is that the library is offering an asset for a given RID, not that it's actually used.

In a sense, that's an actual blind spot in our data gathering and every time we look at NuGet this is a bias we need to consider.